trollmoves/client.py

```diff
@@ -179,21 +174,45 @@ def run(self):
                     beat_monitor(msg)
                     if msg.type == "beat":
                         continue
-
+                    # Handle public "push" messages as a hot spare client
+                    if msg.type == "push":
+                        # TODO: these need to be checked and acted if
+                        # the transfers are not finished on primary
+                        # client and are not cleared
+                        LOGGER.debug("Primary client published 'push'")
+                        add_to_ongoing(msg)
+
+                    # Handle public "ack" messages as a hot spare client
+                    if msg.type == "ack":
+                        LOGGER.debug("Primary client finished transfer")
+                        _ = add_to_file_cache(msg)
+                        _ = clean_ongoing_transfer(get_msg_uid(msg))
+
+                    # If this is a hot spare client, wait for a while
+                    # for a public "push" message which will update
+                    # the ongoing transfers before starting processing here
+                    time.sleep(float(self.ckwargs.get("processing_delay", 0.0)))
```

Comment:
I see a potential bug in here, if many files are being transfered at once (eg when restarting move it server and trying to catch up on accumulated files), stince the sleeps will accumulate.

As an alternative, I would propose using a `CTimer` (from `trollmoves.movers`), like this:
```python
if self.ckwargs.get("processing_delay`, 0):
    timer = CTimer(self.callback, args=[msg] + self.cargs, kwargs=self.ckwargs)
    timer.start()
else:
    self.callback(msg, *self.cargs, **self.ckwargs)
```
You probably want to save the `timer` so that you can interrupt it in `clean_ongoing_transfer`

Reply:
Hmm. Right. I always disable the re-send option on the `server`, so didn't think of that. Also, as in the future (when this hot-spare feature is taken to use) we'll have two independent `server` instances hosting the files, so we shouldn't need resending. I'll have a look at the `CTimer`.